### PR TITLE
[typing/static] assets

### PR DESCRIPTION
**Pull request by smackesey**
### Summary & Motivation

Pure type annotations diff for `AssetGraph` and friends.

### How I Tested These Changes

BK

### CODE REVIEW

**Comment by smackesey** [comment]
Current dependencies on/for this PR:
* master
  * **PR #12003** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12003" target="_blank">Graphite</a>  👈
    * **PR #12004** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12004" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/12003?utm_source=stack-comment).

File: python_modules/dagster/dagster/_utils/cached_method.py [hunk 1]
```diff
@@ -43,18 +51,18 @@ def a_method(self, arg1, arg2):
...
+            cache: Dict[Hashable, T] = {}
             setattr(self, cache_attr_name, cache)
         else:
             cache = getattr(self, cache_attr_name)
 
         key = _make_key(args, kwargs)
         cached_result = cache.get(key, NO_VALUE_IN_CACHE_SENTINEL)
         if cached_result is not NO_VALUE_IN_CACHE_SENTINEL:
-            return cached_result
+            return cached_result  # type: ignore
```
Comment (schrockn) [inline python_modules/dagster/dagster/_utils/cached_method.py:65]: comment-worthy

File: python_modules/dagster/dagster/_utils/cached_method.py [hunk 2]
```diff
@@ -43,18 +51,18 @@ def a_method(self, arg1, arg2):
...
         if not hasattr(self, cache_attr_name):
-            cache = {}
+            cache: Dict[Hashable, T] = {}
             setattr(self, cache_attr_name, cache)
         else:
             cache = getattr(self, cache_attr_name)
 
         key = _make_key(args, kwargs)
         cached_result = cache.get(key, NO_VALUE_IN_CACHE_SENTINEL)
         if cached_result is not NO_VALUE_IN_CACHE_SENTINEL:
```
Comment (schrockn) [inline python_modules/dagster/dagster/_utils/cached_method.py:64]: bit confused why this just isn't:

```python
if key in cache:
   return cache[key]
```

I guess one could argument perf but having the sentinel object seems unnecessarily noisy

**Review by schrockn — APPROVED** [review]
yaytypes

File: python_modules/dagster/dagster/_utils/cached_method.py (hunk 2 above)
Comment (smackesey) [reply python_modules/dagster/dagster/_utils/cached_method.py:64]: that is exactly the change I made upstack in the PR that made the runtime change

**Pull request merged**